hera_cal/omni.py

```diff
@@ -332,9 +332,91 @@ def aa_to_info(aa, pols=['x'], fcal=False, minV=False, **kwargs):
+    solutions. They are put back in at the end. This function only works when there are 
+    exactly 4 visibility polarizations and 2 antenna polarizations. See HERA Memo #30 
+    by Dillon et al. for more details on 4-pol omnical degeneracies at 
+    http://reionization.org/wp-content/uploads/2013/03/HERA30_4PolOmniDegen.pdf
+        
+        Args:
+            info: RedundantInfo object that can parse data
+            g (dict): dictionary of gain solutions (typically from lincal).
+            v (dict): dictionary of model visibilites (typically from lincal).
+            g0 (dict): dictionary of firstcal solutions
+            minV (optional): toggle pseudo-Stokes V minimization.
+
+        Return:
+            g3 (dict): dictionary of gain solutions.
+            v3 (dict): dictionary of model visibilites.
+    '''
+
+    antpols = g.keys()
+    pols = v.keys()
+    antpos = info.get_antpos()
+    if (len(antpols) != 2) or (not (len(pols) == 3 and minV) 
+                               and not (len(pols) == 4 and not minV)):
+        raise ValueError('Trying to use removedegen_4pol without 4-pol data.')
+
+    # Gains loop over antenna fastest, then antpol.
+    ants = {antpol: [ant for ant in g[antpol].keys()] for antpol in antpols}
+    gains = np.array([g[antpol][ant] / g0[antpol][ant] for antpol in antpols 
+                      for ant in ants[antpol]])
+    gainPols = np.array([antpol for antpol in antpols for ant in ants[antpol]])
+    positions = np.array([antpos[ant,0:2] for antpol in antpols 
+                          for ant in ants[antpol]])
+    
+    # Visibilities loop over bl fastest, then pol.
+    bl_pairs = {pol: [bl for bl in v[pol].keys()] for pol in pols}
+    vis = np.array([v[pol][bl_pair] for pol in pols for bl_pair in bl_pairs[pol]])
+    
+    visPols = np.array([[pol[0],pol[1]] for pol in pols for bl_pair in bl_pairs[pol]])
+    bl_vecs = np.array([antpos[i,0:2] - antpos[j,0:2] for pol in pols 
+                        for (i,j) in bl_pairs[pol]])
+
+    # Amplitude renormalization
+    for antpol in antpols:
+        meanAmplitude = np.mean(np.abs(gains[gainPols==antpol]),axis=0)
+        gains[gainPols == antpol] /= meanAmplitude
+        vis[visPols[:,0] == antpol] *= meanAmplitude
+        vis[visPols[:,1] == antpol] *= meanAmplitude
+
+    # Fix phase degeneracies
+    if minV:
+        # In this case, there's only one phase term which should sum to 0
+        Rgains = np.hstack((positions, np.ones((positions.shape[0],1))))
+        Rvis = np.hstack((-bl_vecs, np.zeros((len(bl_vecs),1))))
+    else:
+        # In this case, x and y phases get their own columns (separate degens)
+        phaseTerms = np.vstack((gainPols==antpols[0],gainPols==antpols[1])).T
+        Rgains = np.hstack((positions, phaseTerms))
+        Rvis = np.hstack((-bl_vecs, np.zeros((len(bl_vecs),2))))    
+    Mgains = np.linalg.pinv(Rgains.T.dot(Rgains)).dot(Rgains.T)
```

Comment:
Could you explain what's happening on lines 398 & 399. It's compact and important!